app/fighters/knight.py

```diff
@@ -0,0 +1,37 @@
+class Knight:
+    dict_of_all_knights_instances = {}
+
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: list[dict],
+        weapon: dict,
+        potion: dict,
+    ) -> object:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.dict_of_all_knights_instances[self.name] = self
```

Comment:
Why you need to save it?

Reply:
I'm creating these attributes and storing them with object in dict of all object of the class Knight in order to have possibiity use them when applying perks receiving in perks methods only knight object and also if somebody wants to create the knight object he can do this with all attributes but not by sending, something like this 

otherwise, I guess, my perks methods should receive as an argument base_knights_config and use apply perks values to hp and power, but I've chose maybe bad way(